Add 1.5.17 post (#227)
runelite/runelite.net: src/_posts/2019-03-21-12-00-1.5.17-Release.md

---
title: '1.5.17 Release'
description: 'NPC unaggro timer and chat filter'
author: Adam
---

> A NPC unaggressive timer plugin was added, which shows how long until NPCs near

Reviewer: ```suggestion
A NPC Aggression Timer plugin was added, which shows how long until NPCs near
```

Consistency with what the name of the plugin will be in the client

> you will become unagressive, and how far away you have to walk before they will

Reviewer: ```suggestion
you will become unaggressive, and how far away you have to walk before they will
```

spelling

> become agressive once again.

Reviewer: ```suggestion
become aggressive once again.
```

spelling

---
title: '1.5.17 Release'

> description: 'NPC unaggro timer and chat filter'

Reviewer 2: Same as post body, this should use consistent plugin name.

author: Adam
---

A NPC Aggression Timer plugin was added, which shows how long until NPCs near
you will become unagressive, and how far away you have to walk before they will
become aggressive once again.

![aggro](/img/blog/1.5.17-Release/aggro.png)

The combat level plugin can now show the level range of attackable players near
the wilderness skull icon, similar to how it is on PVP worlds.

![wildylvl](/img/blog/1.5.17-Release/wildylvl.png)

Most coordinate clues now show a description of where they are in addition to
showing on the world map.

![cluedesc](/img/blog/1.5.17-Release/cluedesc.png)

A chat filter plugin was added, which lets you define words and
[patterns](https://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html)
to filter or block from chat messages. With a handful of patterns it is
easy to filter out a majority of the bot spam at the GE.

For example the patterns:

```
5[0-9]x[0-9]
^Player
[0-9.]\s*usd
[0-9][0-9]:[0-9][0-9]:[0-9][0-9]
doubling
tripling
I don't scam
Fire cape & accounts
```

> is highly effecive against the current spam.

Reviewer 2: ```suggestion
is highly effective against the current spam.
```

![chatfilter](/img/blog/1.5.17-Release/chatfilter.png)

We may open up a channel on Discord for users to share patterns similar to bank
tabs.

Even though this is a rather crude solution to the spam epidemic in RS, and

> spamfilters are in general a solved problem, any solution which is much more

Reviewer 2: ```suggestion
spam filters are in general a solved problem, any solution which is much more
```
